### ISSUE TITLE
[Feature request] Recognize objects generated by protobuf

### ISSUE THREAD

**Issue by PhilippSelenium** [PyCQA/pylint#1864]
It would be nice if pylint would recognize members of protobuf objects instead of throwing no-member error.

See also: https://stackoverflow.com/questions/43674028/how-do-you-tell-pylint-what-the-members-of-a-protobuf-generated-object-are

**Comment by PCManticore** [comment]
This is not something that we'll support out of the box. It requires significant efforts to support it, to which we don't have time to allocate. If a patch would come, I would be happy to review and shepard the PR towards the merge, but that's all. An idea would be to approach this by using astroid brain tips, that's probably the best place to go.

**Comment by zapstar** [comment]
Ok, I finally decided to cook something up for this. This is alpha, but I'm willing to spend my energy to take this to completion.

My work tries to implement the design mentioned here [here](https://github.com/PyCQA/pylint/issues/2524#issuecomment-425422843) by @PCManticore and also uses some ideas from [here](https://github.com/nelfin/pylint-protobuf).

The core of the idea is to transform the entire module node and return a new node. Specifically, the generated Protobuf code descriptor object declarations will get transformed into `class` definitions with fields with appropriate types.

I feel that I might be missing something silly, I'm able to generate the class definition for these protobuf messages and enums which I can view via `as_string()` calls, but it is not getting recognized by PyLint.

Here's my prototype: `https://github.com/zapstar/AstroidPlayPen`.

Some helpful pointers on whatever I'm doing wrong or missing would be very helpful.

**Comment by PCManticore** [comment]
This looks like great work @zapstar Thanks for tackling that!

Unfortunately there are no debug flags for a more verbose output from Pylint, what kind of functionality are you looking for in terms of verboseness / debugging?
Regarding why your plugin currently does not work, I confirm it's because of https://github.com/PyCQA/pylint/issues/2878. Ideally we should fix that, but in the meantime if it is possible, you can patch the original node in a similar fashion to how `register_module_extender` changes a module AST by modifying the `locals` of that node.

**Comment by zapstar** [comment]
I copied stuff from `register_module_extender `, and it works!

Also, I was able to copy the contents of the PyLint executable script and get into PyLint code via a debugger, so I'm unblocked on that front as well.

Now I can start working on attribute assignments for these instances:
1. Unknown attribute assignment is NOT allowed.
2. Type checking for existing assignment statements.

Some pointers on approaching this would be very beneficial.
